### PR TITLE
[WFLY-14968] Collapse if-statements (Undertow)

### PR DESCRIPTION
**Pull request by boris-unckel**
Fixes https://issues.redhat.com/browse/WFLY-14968

### CODE REVIEW

**Review by yersan — APPROVED** [review]
LGTM, although component lead could have issues with the cosmetic changes

**Comment by boris-unckel** [comment]
@bstansberry This one has passed tests and is approved, too. Please, could you check and merge?

**Comment by darranl** [comment]
Just be clear regarding approval, the last comment was that the component lead may want an opportunity to review the code for the component they maintain.

**Comment by darranl** [comment]
@fl4via FYI

**Comment by boris-unckel** [comment]
My apologies getting this wrong. I don't want to break any formal or informal rules of review and approval. @darranl Thanks for the clarification. @bstansberry  Please ignore my previous merge request.

**Comment by boris-unckel** [comment]
Good Morning @fl4via , could you review and approve this PR, please? It's one of two missing to finish the Jira and I hope to get it into the release.

File: undertow/src/main/java/org/wildfly/extension/undertow/DiskBasedModularPersistentSessionManager.java [hunk 1]
```diff
@@ -80,10 +80,9 @@ public void start(final StartContext startContext) throws StartException {
             callbackHandle = pathManager.get().registerCallback(pathRelativeTo, PathManager.ReloadServerCallback.create(), PathManager.Event.UPDATED, PathManager.Event.REMOVED);
         }
         baseDir = new File(pathManager.get().resolveRelativePathEntry(path, pathRelativeTo));
-        if (!baseDir.exists()) {
-            if (!baseDir.mkdirs()) {
-                throw UndertowLogger.ROOT_LOGGER.failedToCreatePersistentSessionDir(baseDir);
-            }
+        if (!baseDir.exists()
+                && !baseDir.mkdirs()) {
```
Comment (rhusar) [inline undertow/src/main/java/org/wildfly/extension/undertow/DiskBasedModularPersistentSessionManager.java:84]: Just put these on the same line for readability.

File: undertow/src/main/java/org/wildfly/extension/undertow/deployment/ServletContainerInitializerDeploymentProcessor.java [hunk 2]
```diff
@@ -223,10 +223,9 @@ public void undeploy(final DeploymentUnit context) {
                     if (!servletContainerInitializerClassName.isEmpty()) {
                         // Instantiate the ServletContainerInitializer
                         ServletContainerInitializer service = (ServletContainerInitializer) classLoader.loadClass(servletContainerInitializerClassName).newInstance();
-                        if (service != null) {
-                            if(sciClasses.add(service.getClass())) {
-                                scis.add(service);
-                            }
+                        if (service != null
+                                && sciClasses.add(service.getClass())) {
```
Comment (rhusar) [inline undertow/src/main/java/org/wildfly/extension/undertow/deployment/ServletContainerInitializerDeploymentProcessor.java:227]: Just put these on the same line for readability.

File: undertow/src/main/java/org/wildfly/extension/undertow/deployment/UndertowDeploymentProcessor.java [hunk 3]
```diff
@@ -175,10 +175,10 @@ public void deploy(final DeploymentPhaseContext phaseContext) throws DeploymentU
         final DeploymentUnit deploymentUnit = phaseContext.getDeploymentUnit();
 
         //install the control point for the top level deployment no matter what
-        if(RequestControllerActivationMarker.isRequestControllerEnabled(deploymentUnit)) {
-            if(deploymentUnit.getParent() == null) {
-                ControlPointService.install(phaseContext.getServiceTarget(), deploymentUnit.getName(), UndertowExtension.SUBSYSTEM_NAME);
-            }
+        if (RequestControllerActivationMarker.isRequestControllerEnabled(deploymentUnit)
+                && deploymentUnit.getParent() == null) {
```
Comment (rhusar) [inline undertow/src/main/java/org/wildfly/extension/undertow/deployment/UndertowDeploymentProcessor.java:179]: Just put these on the same line for readability.

File: undertow/src/main/java/org/wildfly/extension/undertow/deployment/UndertowDeploymentProcessor.java [hunk 4]
```diff
@@ -595,11 +595,10 @@ public void undeploy(final DeploymentUnit deploymentUnit) {
 
         //we also register them under the new namespaces
         for (String k : new HashSet<>(ret.keySet())) {
-            if (k != null) {
-                if (k.startsWith(OLD_URI_PREFIX)) {
-                    String newUri = k.replace(OLD_URI_PREFIX, NEW_URI_PREFIX);
-                    ret.put(newUri, ret.get(k));
-                }
+            if (k != null
+                    && k.startsWith(OLD_URI_PREFIX)) {
```
Comment (rhusar) [inline undertow/src/main/java/org/wildfly/extension/undertow/deployment/UndertowDeploymentProcessor.java:599]: Just put these on the same line for readability.

File: undertow/src/main/java/org/wildfly/extension/undertow/security/digest/DigestAuthenticationMechanism.java [hunk 5]
```diff
@@ -261,12 +261,11 @@ public AuthenticationMechanismOutcome handleDigestHeader(HttpServerExchange exch
...
         }
 
-        if (parsedHeader.containsKey(DigestAuthorizationToken.OPAQUE)) {
-            if (!OPAQUE_VALUE.equals(parsedHeader.get(DigestAuthorizationToken.OPAQUE))) {
-                REQUEST_LOGGER.invalidTokenReceived(DigestAuthorizationToken.OPAQUE.getName(),
-                        parsedHeader.get(DigestAuthorizationToken.OPAQUE));
-                return AuthenticationMechanismOutcome.NOT_AUTHENTICATED;
-            }
+        if (parsedHeader.containsKey(DigestAuthorizationToken.OPAQUE)
+                && !OPAQUE_VALUE.equals(parsedHeader.get(DigestAuthorizationToken.OPAQUE))) {
```
Comment (rhusar) [inline undertow/src/main/java/org/wildfly/extension/undertow/security/digest/DigestAuthenticationMechanism.java:265]: Just put these on the same line for readability.

File: undertow/src/main/java/org/wildfly/extension/undertow/security/jacc/WarJACCService.java [hunk 6]
```diff
@@ -307,12 +307,11 @@ public void createPermissions(WarMetaData metaData, PolicyConfiguration pc) thro
...
             final int NUMBER_OF_HTTP_METHODS = 7;
             // JACC 1.1: create !(httpmethods) in unchecked perms
-            if(jbossWebMetaData.getDenyUncoveredHttpMethods() == null) {
-                if (seenMethods.size() != NUMBER_OF_HTTP_METHODS) {
-                    WebResourcePermission wrpUnchecked = seenMethods.isEmpty() ? new WebResourcePermission(qurl, (String) null)
-                            : new WebResourcePermission(qurl, "!" + getCommaSeparatedString(seenMethods.toArray(new String[seenMethods.size()])));
-                    pc.addToUncheckedPolicy(wrpUnchecked);
-                }
+            if(jbossWebMetaData.getDenyUncoveredHttpMethods() == null
+                && seenMethods.size() != NUMBER_OF_HTTP_METHODS) {
```
Comment (rhusar) [inline undertow/src/main/java/org/wildfly/extension/undertow/security/jacc/WarJACCService.java:311]: Just put these on the same line for readability.

**Review by rhusar — CHANGES_REQUESTED** [review]

**Comment by boris-unckel** [comment]
@rhusar Hi Radoslav, thanks for your feedback. It's fixed. Could you review again, please?
@fl4via fyi

**Review by fl4via — APPROVED** [review]

**Comment by fl4via** [comment]
@boris-unckel thanks for reopening! @bstansberry  this one is good to go

**Pull request merged**